src/tilesource.js

```diff
@@ -275,28 +275,43 @@ $.TileSource.prototype = {
         return this._tileHeight;
     },
 
+    /**
+     * Set the maxLevel to the given level, and perform the memoization of
+     * getLevelScale with the new maxLevel. This function can be useful if the
+     * memoization is required before the first call of getLevelScale, or both
+     * memoized getLevelScale and maxLevel should be changed accordingly.
+     * @function
+     * @param {Number} level
+     */
+    setMaxLevel: function( level ) {
+        this.maxLevel = level;
+        this._memoizeLevelScale();
+    },
+
     /**
      * @function
      * @param {Number} level
      */
     getLevelScale: function( level ) {
+        // if getLevelScale is not memoized, we generate the memoized version
+        // at the first call and return the result
+        this._memoizeLevelScale();
+        return this.getLevelScale( level );
+    },
 
+    // private
+    _memoizeLevelScale: function() {
         // see https://github.com/openseadragon/openseadragon/issues/22
         // we use the tilesources implementation of getLevelScale to generate
         // a memoized re-implementation
-        var maxLevel = Math.ceil(
-                Math.log( Math.max( this.dimensions.x, this.dimensions.y ) ) /
-                Math.log( 2 )
-            ),
-            levelScaleCache = {},
+        var levelScaleCache = {},
             i;
-        for( i = 0; i <= maxLevel; i++ ){
-            levelScaleCache[ i ] = 1 / Math.pow(2, maxLevel - i);
+        for( i = 0; i <= this.maxlevel; i++ ){
```

Comment:
I tried running this to test it, and it didn't work. After some debugging, I added `var maxLevel = this.maxLevel` above the for loop and then replaced both of the `this.maxLevel` in the for loop with `maxLevel` and that fixed it. It seems very strange to me that that would make any sort of difference but it seems to be important! The only thing I can think of is maybe there's something weird happening with the code minification that's causing it to make a difference. 

Are you experiencing the same weirdness? At any rate, please add that fix.

Reply:
Also it would probably be good to add a comment to explain why this "fix" is here, to warn future developers.

Reply:
Thank you for the double-check. 🤦‍♂️ I found that I've made an embarrassing mistake... I tried to access `this.maxlevel` not `this.maxLevel`, and fixing the typo seems to make it working. Also, I modified the unit test based on the `setMaxLevel` functionality. I am really sorry for making your time spent on my fault.

Reply:
Oh! I'm so glad there's a reasonable explanation! I thought I was going crazy, but for some reason I just didn't notice the lowercase L. No worries... I'm glad it got sorted out :)